[sim zhe feng kenneth]

Kennethsim2000 opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen

Duke frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/Kennethsim2000/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines 
- [x] Filter tasks

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public static void main(String[] args)  {
        try {
            new Duke().DukeTask();
        } catch (DukeException e) {
            System.out.println(e.getMessage());
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            System.out.println("Path name cannot be found");
            //e.printStackTrace();
        }
    }
```

Kennethsim2000 committed:
Completed level 1

Kennethsim2000 committed:
completed level 2

Kennethsim2000 committed:
Completed level 3

Kennethsim2000 committed:
Completed level 4

Kennethsim2000 committed:
Level 5-handle error done

Kennethsim2000 committed:
Completed level 6

Kennethsim2000 committed:
edited level 6

Kennethsim2000 committed:
cleaned up code

Kennethsim2000 committed:
Done testing

Kennethsim2000 committed:
Complete Level 7

Kennethsim2000 committed:
Revert "Complete Level 7"

This reverts commit 7dbaf7fb2350c671ad03c0a8514e9af240cbef16.

Kennethsim2000 committed:
Revert "Revert "Complete Level 7""

This reverts commit e1f6b685e6bbd0a13b9890f67a52ff9c8eaf2c64.

Kennethsim2000 committed:
Complete Level-7

Kennethsim2000 committed:
Revert "Complete Level 7"

This reverts commit 7dbaf7fb2350c671ad03c0a8514e9af240cbef16.

Kennethsim2000 committed:
Merge branch 'branch-Level-7'

Kennethsim2000 committed:
Complete Level-8

Kennethsim2000 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Kennethsim2000 committed:
Remove comments

Kennethsim2000 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Kennethsim2000 committed:
Complete Level-8

Kennethsim2000 committed:
Extract Duke.java into 4 classes
Parser.java
Storage.java
TaskList.java
Ui.java

Kennethsim2000 committed:
Add gradle

Kennethsim2000 committed:
Organize classes into packages

Kennethsim2000 committed:
Add Search functionality

Kennethsim2000 committed:
Add mark and unmark

Kennethsim2000 committed:
Edit code standard

Kennethsim2000 committed:
Create jar file

Kennethsim2000 committed:
Finish tests

Kennethsim2000 committed:
Complete javadoc.

Kennethsim2000 committed:
Change string arraylist to task

ongwenpin commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,36 @@
+     */
+    @Override
+    public String toString() {
+        return "["+ getStatusIcon()+"] "+this.description;
```
Perhaps you have forgotten to add space between the + ?

ongwenpin commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,146 @@
+                System.out.println(task);
+                scanner.nextLine();
+            } else if (str.equals("mark")) {
+                String strnum = scanner.next();
+                int num = Integer.valueOf(strnum);
+                Task task;
+                if (num > tasklist.oldTasksSize()) {
+                    task = tasklist.getNewTasks(num-tasklist.oldTasksSize()-1);
+                    task.markDone();
+                    tasklist.setNewTasks(num-tasklist.oldTasksSize()-1, task);
+                } else {
+                    task = tasklist.getOldTasks(num-1);
+                    task.markDone();
+                    tasklist.setOldTasks(num-1, task);
+                }
+                bot.markTask(true);
+                System.out.println(task);
+                scanner.nextLine();
+            }
+            else{
+                if (tasklist.oldTasksSize() < initialSize) {
+                    storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                } else {
+                    storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                    //storage.addTasks("data/Duke2.txt",tasklist.newTasks);
+                }
+                bot.displayError();
+            }
+
+        }  while (!str.equals("bye"));
+        if (str.equals("bye")) {
+            if(tasklist.oldTasksSize() < initialSize){
+                storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+            } else{
+                storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                //storage.addTasks("data/Duke2.txt",tasklist.newTasks);
+            }
+            bot.goodBye();
+        }
+   }
```
Perhaps it is better to make multiple shorter methods for easier readability and future reference?

Kennethsim2000 replied:
oh ya changed it alr thanks!!

ongwenpin reviewed (commented):
Overall, code looks good and well organised, LGTM!

Kennethsim2000 committed:
Level-10

Kennethsim2000 committed:
Completed Level 10

lulucopter commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,105 @@
+//import javafx.geometry.Pos;
+//import javafx.scene.layout.HBox;
+//
+//public class DialogBox extends HBox{
+//    private Label text;
+//    private ImageView displayPicture;
+//
+//    public DialogBox(Label l, ImageView iv) {
+//        text = l;
+//        displayPicture = iv;
+//
+//        text.setWrapText(true);
+//        displayPicture.setFitWidth(100.0);
+//        displayPicture.setFitHeight(100.0);
+//
+//        this.setAlignment(Pos.TOP_RIGHT);
+//        this.getChildren().addAll(text, displayPicture);
+//    }
+//
+//    /**
+//     * Flips the dialog box such that the ImageView is on the left and text on the right.
+//     */
+//    private void flip() {
+//        this.setAlignment(Pos.TOP_LEFT);
+//        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
+//        FXCollections.reverse(tmp);
+//        this.getChildren().setAll(tmp);
+//    }
+//
+//    public static DialogBox getUserDialog(Label l, ImageView iv) {
+//        return new DialogBox(l, iv);
+//    }
+//
+//    public static DialogBox getDukeDialog(Label l, ImageView iv) {
+//        var db = new DialogBox(l, iv);
+//        db.flip();
+//        return db;
+//    }
+//
+//}
```
Tip for commenting large blocks of code! You can save some time by closing your comment block with `/*` and `*/` instead of double slashing every line

lulucopter commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+package Duke;
```
Good job packaging all your classes into a package! Remember that package names should be in lower case

lulucopter commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * Represents a task that will occur on a particular date
+ */
+
```
Great to see javadocs for public classes and methods! You might have missed this out accidentally, but remember that there should not be any blank line between your header comment and your class/method

ongwenpin replied:
You're right, thank you for pointing out

lulucopter commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,28 @@
+
+    /**
+     * Returns the string representation of the event
+     * @return String representation of the task.
```
There should be a line spacing between the parameters and the description as well!

lulucopter commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,134 @@
+    public Parser(TaskList tasklist, Ui bot, Storage storage) {
+        this.tasklist = tasklist;
+        this.bot = bot;
+         this.storage = storage;
```
Perhaps you typed a space by accident?

lulucopter commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,134 @@
+            }
+        } else if (arr[0].equals("event")) {
+            String description = "";
+            String time="";
```
Don't forget to space out operands

lulucopter commented on src/main/java/Duke/ToDo.java:
```diff
@@ -0,0 +1,11 @@
+package Duke;
+
```
Perhaps you'd forgotten to include javadocs here?

lulucopter reviewed (commented):
Overall LGTM, other than some small nits! Great Job! 
Liked how you followed OOP principles on immutability, especially for the TaskList class.

Kennethsim2000 committed:
Some classes Event, Parser, Task and ToDo do not entirely follow
code quality

Not following code quality may affect readability which is not good
for other developers reading my code.

lets rectified the format of some parts to ensure that code quality
is followed.

Following code quality will ensure that my code is more readable.

Kennethsim2000 committed:
Deleted irrelevant code

Kennethsim2000 committed:
Tasklist currently uses two lists, oldlist and newlist

Two tasklist can be confusing to work on especially when doing
adds and deletions.

Let's combine the two tasklists to one tasklist and perform
operations on one tasklist instead.

Provides easier convenience of managing tasklist in the future

Kennethsim2000 committed:
Add sort by deadline to bot.

Kennethsim2000 committed:
Add assertion to todo class.

Kennethsim2000 committed:
Merge pull request #1 from Kennethsim2000/branch-A-Assertions

Add assertion to todo class.

Kennethsim2000 committed:
Fix coding standard

Kennethsim2000 committed:
Rename package

Kennethsim2000 committed:
Merge pull request #3 from Kennethsim2000/branch-A-CodeQuality

Rename package

Kennethsim2000 committed:
Rename package Duke and resolve errors.

Kennethsim2000 committed:
Modify file creation.

Kennethsim2000 committed:
Update read.me.

Kennethsim2000 committed:
Update READ.ME

Kennethsim2000 committed:
A-UserGuide

Kennethsim2000 committed:
Add screenshot

Kennethsim2000 committed:
Merge branch 'master' of https://github.com/Kennethsim2000/ip

Kennethsim2000 committed:
Modify build.gradle

Kennethsim2000 committed:
Rename package

Kennethsim2000 committed:
Complete abstraction

Kennethsim2000 committed:
Abstract out storage

Kennethsim2000 committed:
Update README.md

Kennethsim2000 committed:
Update README.md

Kennethsim2000 committed:
Update README.md

Kennethsim2000 committed:
Update README.md

Kennethsim2000 committed:
Handle improper input.

Kennethsim2000 committed:
Include description proper exception
